exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/scan/file/FileScanFramework.java

```diff
@@ -206,7 +206,13 @@ protected FileSplit nextSplit() {
 
   @Override
   protected SchemaNegotiatorImpl newNegotiator() {
-    return new FileSchemaNegotiatorImpl(this);
+    SchemaNegotiatorImpl nego = new FileSchemaNegotiatorImpl(this);
+    String tableName = Path.getPathWithoutSchemeAndAuthority(currentSplit.getPath()).toString();
+    if (currentSplit.getStart() != 0) {
+      tableName += String.format(", offset %,d", currentSplit.getStart());
+    }
+    nego.setTableName(tableName);
```

Comment:
Not quite sure what table name this will be, why you need offset etc. Can you please give an example?